src/pages/Items/AddItem/AddItem.tsx

```diff
@@ -33,13 +46,56 @@ const AddItem = () => {
           [name]: true,
         };
       });
+      setInputValues((prev) => ({
+        ...prev,
+        [name]: currentValue,
+      }));
     } else if (currentValue.length < 1) {
       setIsValid((prev) => {
         return {
           ...prev,
           [name]: false,
         };
       });
+      setInputValues((prev) => ({
+        ...prev,
+        [name]: currentValue,
+      }));
+    }
+  };
+
+  const imgFileUpload = async (imgFile: FormData) => {
+    const result = await postUploadImage(imgFile);
+    if (result) {
+      setInputValues((prev) => ({
+        ...prev,
+        images: result.url,
+      }));
+    }
+  };
+
+  const uploadPostMutation = useMutation({
+    mutationFn: (newPost: postProduct) => postAddItem(newPost),
+    onSuccess: (data) => {
+      const id = data.id;
+      navigate(`/items/${id}`);
+    },
```

Comment:
data의 id 속성만 활용한 가능성이 높은 부분이라 아래와 같이 작성해도 좋을 것 같습니다.

```suggestion
    onSuccess: ({ id }) => {
      navigate(`/items/${id}`);
    },
```

Reply:
만약, 확장성 있게 작성하려면 변수 선언 시 구조 분해 할당을 사용해도 좋을 것 같습니다

```suggestion
    onSuccess: (data) => {
      const { id } = data
      navigate(`/items/${id}`);
    },
```